app/services/catalog/create_request_body_from.rb

```diff
@@ -25,6 +26,16 @@ def process
 
     private
 
+    def platform(portfolio_item)
+      service_offering = TopologicalInventory.call do |api_instance|
+        api_instance.show_service_offering(portfolio_item.service_offering_ref)
+      end
```

Comment:
Can we skip this block by using the `service_offering_source_ref` property of portfolio_item in the block below?

Reply:
Potentially. I built this based on what I could do via the API. I will check again.

Reply:
Unless I'm reading the code wrong you are doing the `service_offering` lookup to get the `source_id` and the `source_id` value is already stored in `portfolio_item.service_offering_source_ref`.  And `service_offering_source_ref` is available from the portfolio_item API.